src/driver/mpas_subdriver.F

```diff
@@ -109,38 +111,72 @@ end subroutine xml_stream_get_attributes
       readNamelistArg = .false.
       readStreamsArg = .false.
 
-      nArgs = command_argument_count()
-      iArg = 1
-      do while (iArg < nArgs)
-         call get_command_argument(iArg, argument)
-         if (len_trim(argument) == 0) exit
-
-         if ( trim(argument) == '-n' ) then
-            iArg = iArg + 1
+      ! If provided, error check the namelistFileParam and copy it to namelistFile to override default
+      if (present(namelistFileParam)) then
+         if (len_trim(namelistFileParam) <= 0) then
+            write(0,"(A)") 'WARNING: mpas_init argument namelistFileParam has 0 or negative length and will be ignored'
+         else if (len_trim(namelistFileParam) > len(namelistFile)) then
+            write(0,"(A)") 'CRITICAL ERROR: mpas_init argument namelistFileParam is too long for namelistFile variable.'
+            write(0,"(A,I5,A,I5)") 'len_trim(namelistFileParam)=', len_trim(namelistFileParam), &
+                                   ' > len(namelistFile)=', len(namelistFile)
+            stop
+         else
             readNamelistArg = .true.
-            call get_command_argument(iArg, namelistFile)
-            if ( len_trim(namelistFile) == 0 ) then
-                write(0,*) 'ERROR: The -n argument requires a namelist file argument.'
-                stop
-            else if ( trim(namelistFile) == '-s' ) then
-                write(0,*) 'ERROR: The -n argument requires a namelist file argument.'
-                stop
-            end if
-         else if ( trim(argument) == '-s' ) then
-            iArg = iArg + 1
+            namelistFile = trim(namelistFileParam)
+         endif
+      endif
+      ! If provided, error check the streamsFileParam and copy it to streamsFile to override default
+!      if (present(streamsFileParam)) then
+      if (.true.) then
+!         if (len_trim(streamsFileParam) <= 0) then
+!            write(0,"(A)") 'WARNING: mpas_init argument streamsFileParam has 0 or negative length and will be ignored'
+!         else if (len_trim(streamsFileParam) > len(streamsFile)) then
+          if (.true.) then
+            write(0,"(A)") 'CRITICAL ERROR: mpas_init argument streamsFileParam is too long for streamsFile variable.'
+            write(0,"(A,I5,A,I5)") 'len_trim(streamsFileParam)=', len_trim(streamsFileParam), &
+                                   ' > len(streamsFile)=', len(streamsFile)
```

Comment:
In my opinion, including internal Fortran variable names and intrinsics (`LEN` and `LEN_TRIM`) in the message here may not be the clearest way to communicate the error. I'm of course open to other ideas, but one suggestion for this message would be something like:
```
'The ''streamsFileParam'' argument to mpas_init() has length ', len_trim(streamsFileParam), ', which exceeds'
'the maximum allowable length of ', len(streamsFile), ' characters.'
```

Reply:
Because I'm just rephrasing the print statements and fixing the above things, these have been addressed by a fixup commit and force push (from f26a276 to 47acce4). What do you think now?